persistence/README.md

```diff
@@ -78,232 +114,306 @@ This document presents a specification for the Persistence Module of the Pocket
+- When updating an existing structure, all uniquely-identified fields must be indicated explicitly in the operation. Specifically, timestamp fields such as **Created At (created_at)** and **Deleted At (deleted_at)** must also be specified explicitly to avoid collisions and discrepancies between the structure's data and metadata.
+- When deleting a structure, all uniquely-identified fields must be indicated in the operation. In addition, deletion through filters or via **"all"** will not be permitted by the system.
 
 ### 2.3. Deterministic Write mechanism
 
-Most modern SQL DBMS (database management system) implementations utilize the **Transaction model** to indicate a transaction as a group of operations to be performed sequentially and logically grouped. This is called [ACID](https://en.wikipedia.org/wiki/ACID) (atomicity, consistency, isolation, durability), which are the properties that allow the dataset to stay consistent across operations, by allowing capabilities such as the rollbacks in case one of the operations in the transaction fails or yields an inconsistent dataset state.
+Most modern SQL DBMS (database management system) implementations utilize the **Transaction model** to indicate a transaction as a group of operations to be performed sequentially and logically grouped. This is called **[ACID]**(https://en.wikipedia.org/wiki/ACID) (atomicity, consistency, isolation, durability), which are the properties that allow the dataset to stay consistent across operations by enabling capabilities such as rollbacks in case one of the operations comprising an ACID transaction fails or yields an inconsistent dataset state.
 
-Our chosen database engine, Postgresql, defines a transaction model in their official documentation [here](https://www.postgresql.org/docs/current/tutorial-transactions.html). On top of that the capacity to rollback transactions is described [here](https://www.postgresql.org/docs/current/sql-rollback-prepared.html). In combination these 2 mechanisms allow us to establish deterministic writes, which will allow the handling of datasets such as **state** in a fail-safe manner avoiding issues like data corruption and race conditions. 
+Our chosen database engine, **Postgresql**, defines a transaction model in its official documentation [here](https://www.postgresql.org/docs/current/tutorial-transactions.html). It also defines the ability to rollback transactions [here](https://www.postgresql.org/docs/current/sql-rollback-prepared.html). These mechanisms establish the ability to achieve deterministic writes while avoiding issues such as data corruption and race conditions.
 
+<!-- TODO(discuss in review): 2.2.1.3 and 2.3 are very similar. Confirm that both are really needed and that the number is correct (because the prod version is a bit misconfigured).-->
```

Comment:
The difference between the 2 is that in `2.2.1.3` we specify the need that if a set of information would be "deleted" or "inserted" that it gets "upserted" yielding net-zero change to the data. In 2.3 we specify the need that individual changes to the state machine get batched in `1 transaction` to allow them to be atomic and deterministic, meaning that once the dataset is updated, it's updated as a whole and not just partially.

Reply:
`we specify the need that individual changes to the state machine get batched in 1 transaction`

Yea, this makes sense now, especially after we've taken a first stab at the implementation.

The key difference in the implementation is that every blockchain transaction we apply turns into a single atomic DB transaction, which is why we ultimately needed to implement the "ON CONFLICT DO UPDATE" approach (i.e. atomic across different transactions but within a single height). I think this is an implementation detail, but ultimately what we're doing is leveraging the DB engine logic for conflicts rather than aggregating something ourselves.

I'm thinking we can leave it as is, an in a next iteration I could discuss `For example, a possible implementation of this using postgres is ...`. Wdyt?